setter/setter.go

```diff
@@ -53,6 +53,54 @@ func SetDefault(dest interface{}, defaultValue ...interface{}) {
 	}
 }
 
+// Default assigns the first non-zero default value to `dest`
+// if `dest`, itself, is the zero value of type T.
+// To work with slices: use DefaultSlice
+//
+//	 var config struct {
+//			Verbose *bool
+//			Foo string
+//			Bar int
+//		}
+//		holster.Default(&config.Foo, "default")
+//		holster.Default(&config.Bar, 200)
+//
+// Supply additional default values and SetDefault will
+// choose the first default that is not of zero value
+//
+//	holster.SetDefault(&config.Foo, os.Getenv("FOO"), "default")
+func Default[T comparable](dest *T, defaultValues ...T) {
+	if IsZeroNew(*dest) {
+		for _, value := range defaultValues {
+			if !IsZeroNew(value) {
+				*dest = value
+				return
+			}
+		}
+	}
+}
+
+// DefaultSlice assigns the first non-empty default value to `dest` if
+// `dest`, itself, is an empty slice.
+func DefaultSlice[T comparable](dest *[]T, defaultValues ...[]T) {
+	if len(*dest) == 0 {
+		for _, value := range defaultValues {
+			if len(value) != 0 {
+				*dest = make([]T, len(value))
+				copy(*dest, value)
+				return
+			}
+		}
+	}
+}
+
+// IsZeroNew compares the given value to its Golang-specified zero value.
+// It works for any type T that satisfies comparable.
+func IsZeroNew[T comparable](value T) bool {
```

Comment:
should it be exportable?

Reply:
Good question. The original method is both exported and used in a few libraries so that was the motivation, though I'm still pondering if it's needed 🤔

Reply:
It is better to make it non-exportable and who needs to copy it as it's pretty small:

> A little copying is better than a little dependency.

https://go-proverbs.github.io/

Well, I just don't like this name 😄

Reply:
Bahaha I don't like it either :D I'll leave the existing method (IsZero) exported and un-export this one, given the restricted use 👍